flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/RecordDeserializer.java

```diff
@@ -63,4 +65,13 @@ public boolean isBufferConsumed() {
 	void clear();
 
 	boolean hasUnfinishedData();
+
+	/**
+	 * Gets the unconsumed buffer which needs to be persisted in unaligned checkpoint scenario.
+	 *
+	 * <p>Note that the unconsumed buffer might be null if the whole buffer was already consumed
+	 * before and there are no partial length or data remained in the end of buffer.
+	 */
+	@Nullable
+	Buffer getUnconsumedBuffer();
```

Comment:
Again, maybe use Optional?

Reply:
For API/public methods, it is indeed to suggest using `Optional` in practice. I would adjust it.